### DISCUSSION TITLE
Apache 2.0 License is incompatible with GPL v2

### DISCUSSION THREAD

**Discussion by dralley** [aio-libs/aiohttp#6514] [Ideas]
### Describe the bug

It appears that this library used to be licensed as MIT and eventually switched to Apache 2.0 for patent protections: https://github.com/aio-libs/aiohttp/issues/175

This is understandable but unfortunate because the Apache 2.0 license is not compatible with the GPL v2 license (at least according to the FSF, opinions on this differ).

The standard workaround for this (very common in the Rust community) is to license projects as MIT *OR* Apache 2.0, which means that contributions invoke the Apache 2.0 protections but downstream users can choose which license they want to accept the code under.  

It would be excellent if aiohttp followed this strategy.

### To Reproduce

N/A

### Expected behavior

N/A

### Logs/tracebacks

```python-traceback
N/A
```

### Python Version

```console
N/A
```

### aiohttp Version

```console
$ python -m pip show aiohttp
```

### multidict Version

```console
$ python -m pip show multidict
```

### yarl Version

```console
$ python -m pip show yarl
```

### OS

N/A

### Related component

Server

### Code of Conduct

- [X] I agree to follow the aio-libs Code of Conduct

**Comment by asvetlov** [comment]
The switch happened in 2014, when aiohttp was in early beta.
Since that time I don't recall any license blame. Seems like people are satisfied with the status quo.

IIRC Apache 2.0 is compatible with GPLv3. GPLv3 is not compatible with older GPLv2.
Why should we care? 
Why is GPLv2 support so important? 
Should GPLv1 be supported?
Is Python PSF license considered GPLv2 compatible? Why?

I'm not a lawyer though.

**Reply by dralley** [reply]
Frankly most people don't pay much attention to licensing issues :shrug: 

re: GPLv2, I don't have any statistics but I honestly expect it's more widespread than GPLv3.  As far as I can tell the PSF license is compatible.

**Reply by Dreamsorcerer** [reply]
> re: GPLv2, I don't have any statistics but I honestly expect it's more widespread than GPLv3.

I was under the impression the majority of projects were GPLv2+ or GPLv3. The Linux kernel being the only notable exception that comes to mind as GPLv2 only.

**Reply by dralley** [reply]
Is "GPLv2+" fine in combination with Apache-2.0 though?  IANAL

**Reply by Dreamsorcerer** [reply]
Well, yes, because if you have a project under GPLv2+ and another project under something not compatible with GPLv2, then you just release the combined work under GPLv3 (or GPLv3+)...

**Comment by diegomontanopsn-max** [comment]
atisfied with the status quo, which is why maintainers rarely reopen relicensing discussions unless a major enterprise stakeholder requires it.

Dual-licensing (MIT OR Apache 2.0) remains the most pragmatic path forward for open-source libraries that want both patent defenses and maximum compatibility across GPLv2, GPLv3, and commercial ecosystems.

**Reply by webknjaz** [reply]
Such things are normally fine to decide at the beginning of the project. But when it's already established, there's no practical way of adding something incompatible — every prior contributor (with a substantial/copyrightable change accepted) would have to be tracked down and asked to agree.
